core/lib/zksync_core/src/api_server/web3/namespaces/zks.rs

```diff
@@ -571,4 +571,18 @@ impl ZksNamespace {
             .base_token_address
             .ok_or(Web3Error::NotImplemented)
     }
+
+    #[tracing::instrument(skip(self))]
+    pub async fn get_gas_per_pubdata_byte_impl(&self) -> U256 {
```

Comment:
Dumb question: Would it be unreasonable to have a method returning `PubdataIndependentBatchFeeModelInput`, which contains data necessary to compute the required value, and also would obsolete the adjacent `zks_getL1GasPrice` method? IMO, it's better to have fewer methods with greater utility than a multitude of overspecialized methods.

Reply:
Great point, I did exactly that. Please, have a look and check whether it's what you were thinking